File: kotlin/0787-cheapest-flights-within-k-stops.kt
class Solution {
    fun findCheapestPrice(n: Int, flights: Array<IntArray>, src: Int, dst: Int, k: Int): Int {
        var cost = IntArray(n){ Int.MAX_VALUE }
        cost[src] = 0

        for (i in 0..k) {
            val temp = cost.clone()

            for (flight in flights) {
                val (u, v, w) = flight

                if (cost[u] != Int.MAX_VALUE)
                    temp[v] = Math.min(temp[v], cost[u] + w)
            }

            cost = temp
        }

        return if (cost[dst] == Int.MAX_VALUE) -1 else cost[dst]
    }
}

File: go/0787-cheapest-flights-within-k-stops.go
func findCheapestPrice(n int, flights [][]int, src int, dst int, k int) int {
	cost := make([]int, n)

	for i := 0; i < n; i++ {
		cost[i] = math.MaxUint32
	}

	cost[src] = 0

	for i := 0; i <= k; i++ {
		temp := make([]int, n)
		copy(temp, cost)

		for j := 0; j < len(flights); j++ {
			from, to, price := flights[j][0], flights[j][1], flights[j][2]

			if cost[from] != math.MaxUint32 {
				temp[to] = min(temp[to], cost[from]+price)
			}
		}

		cost = temp
	}

	if cost[dst] == math.MaxUint32 {
		return -1
	}

	return cost[dst]
}

func min(x, y int) int {
	if x > y {
		return y
	}

	return x
}